runtime/autoload/remote/host.vim

```diff
@@ -118,36 +118,86 @@ function! remote#host#RegisterPlugin(host, path, specs) abort
 endfunction
 
 
-function! s:GetManifest() abort
+" Get the path to the rplugin manifest file and ensure the directory is
+" writeable.  Preferred locations are checked before falling back to sensible
+" locations.
+function! s:GetManifestPath() abort
+  let manifest_base = ''
+
+  if has('win32') || has('win64')
+    let preferred = ['$LOCALAPPDATA', '~/AppData/Local']
+  else
+    let preferred = [
+          \ '$XDG_DATA_HOME', '~/.local/share',
+          \ '$XDG_CONFIG_HOME', '~/.config']
+  endif
+
+  " Try the preferred locations first.
+  for dest in preferred
+    if !exists(dest)
+      continue
+    endif
+
+    let dest = fnamemodify(expand(dest), ':p')
+    if !empty(dest) && isdirectory(dest) && filewritable(dest) == 2
+      let dest .= 'nvim/'
+
+      if isdirectory(dest) && filewritable(dest) != 2
+        continue
+      endif
+
+      call mkdir(dest, 'p', 700)
+      let manifest_base = dest
+      break
+    endif
+  endfor
+
+  " Try the $MYVIMRC directory.
+  if empty(manifest_base) && exists('$MYVIMRC') && !empty($MYVIMRC)
+    let manifest_base = fnamemodify($MYVIMRC, ':p:h').'/'
+    if filewritable(manifest_base) != 2
+      let manifest_base = ''
+    endif
+  endif
+
+  " Final resort.
+  if empty(manifest_base)
+    let manifest_base = fnamemodify('~', ':p:h').'/'
+  endif
+
+  return manifest_base.'.nvim_rplugin.vim'
```

Comment:
It should never need MYVIMRC any more. If XDG variable is empty, it creates `~/.local/share/nvim` an uses that. Just like backup, undodir, shada, etc.